control-plane-operator/main.go

```diff
@@ -513,3 +488,7 @@ func NewStartCommand() *cobra.Command {
 
 	return cmd
 }
+
+func isAWS() bool {
+	return os.Getenv("AWS_REGION") != ""
```

Comment:
Should this be:
```
func isAWSPrivate() bool {
	return os.Getenv("AWS_REGION") != "" && hcp.Spec.Platform.Type == hyperv1.AWSPlatform
}
```

Reply:
We are no longer retrieving the hcp on startup, but we're always setting AWS_REGION on the CPO when platform.Type == AWS:
https://github.com/openshift/hypershift/blob/a3791d6fe5949fa50a3a7109893e54097ad7d496/hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go#L2739-L2770 
So just retrieving the hcp on startup for the same check seemed unnecessary

Reply:
ah I see. Where I was actually trying to get is that I guess we should only run this if we are in aws private, i.e. should we check hcp.spec.EndpointAccess?

Reply:
Yup, we can. Since we need the hcp for that, I'll restore the code that fetched it on startup.